backend/src/controllers/dish.ts

```diff
@@ -100,29 +74,70 @@ export const getDishes = async (req: Request, res: Response) => {
             message: 'Error when mapping dishes to transactions',
             statusCode: 500,
         })
-        res.status(500).json({ error: 'internal_server_error' })
-        return
+        return res.status(500).json({ error: 'internal_server_error' })
+    }
+
+    let userDishes
+    if (withTransactions !== 'true') {
+        try {
+            if (borrowed == 'true') {
```

Comment:
Can you do a strict check?
```suggestion
            if (borrowed === 'true') {
```